### PR TITLE
add animated coin icons and text fallbacks

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1455

## What changed (plus any additional context for devs)
- converted swap coin icon to be use AnimatedFasterImage impl.
- coverted fallback to be an Animated.View

## What is left
- [x] animated chain badge needs to use local png data or serve those from an API

### CODE REVIEW

File: src/__swaps__/screens/Swap/components/AnimatedChainImage.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    console.log(url);
+    if (url) {
+      base.source.url = url;
+      return base;
+    }
+
+    // fallback to static network badge data
+    base.source.url = `file://${networkBadges[asset.value.chainId]}`;
+
+    console.log(base);
```
Comment (christianbaroni) [inline src/__swaps__/screens/Swap/components/AnimatedChainImage.tsx:85]: log

**Review by christianbaroni — COMMENTED** [review]
question: is your plan to use these animated coin/chain icon components exclusively for the selected input and output assets? (I think that makes sense, just confirming)

**Comment by walmat** [comment]
> question: is your plan to use these animated coin/chain icon components exclusively for the selected input and output assets? (I think that makes sense, just confirming)

yep. I guess I should bring back the other version for the coinrow icons. Forgot about that, good call.

File: src/__swaps__/screens/Swap/components/SwapSheetGestureBlocker.tsx [hunk 2]
```diff
@@ -13,28 +13,28 @@ export const SwapSheetGestureBlocker = ({
   children: React.ReactNode;
   preventScrollViewDismissal?: boolean;
 }) => {
-  const { configProgress, inputProgress, outputProgress } = useSwapContext();
-  const [enabled, setEnabled] = useState<boolean>(false);
+  // const { configProgress, inputProgress, outputProgress } = useSwapContext();
```
Comment (benisgold) [inline src/__swaps__/screens/Swap/components/SwapSheetGestureBlocker.tsx:16]: is this stuff meant to remain commented out?

File: src/__swaps__/screens/Swap/components/AnimatedChainImage.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      return base;
+    }
+
+    const url = getCustomChainIconUrlWorklet(asset.value.chainId, asset.value.address);
+    if (url) {
+      base.source.url = url;
+      return base;
+    }
+
+    // TODO: How can we reference local static pngs here?
```
Comment (benisgold) [inline src/__swaps__/screens/Swap/components/AnimatedChainImage.tsx:77]: chatgpt says you'd have to do something like this 

```
const networkBadges = {
  1: require('./assets/networkBadge1.png'),
  2: require('./assets/networkBadge2.png'),
  // Add more mappings as needed
};

const getLocalImageUrl = (chainId) => {
  switch (chainId) {
    case 1:
      return Image.resolveAssetSource(networkBadges[1]).uri;
    case 2:
      return Image.resolveAssetSource(networkBadges[2]).uri;
    // Add more cases as needed
    default:
      return '';
  }
};
```

File: src/__swaps__/screens/Swap/components/SwapSheetGestureBlocker.tsx (hunk 2 above)
Comment (walmat) [reply src/__swaps__/screens/Swap/components/SwapSheetGestureBlocker.tsx:16]: yeah disabled until https://linear.app/rainbow/issue/APP-1459/implement-swapsheetgestureblocker is in

**Review by benisgold — APPROVED** [review]

**Pull request merged**